src/python_bindings/mod.rs

```diff
@@ -80,6 +91,52 @@ impl PyGuide {
         }
     }
 
+    /// Rollback the Guide state `n` tokens (states).
+    /// Fails if `n` is greater than stored prior states.
+    fn rollback(
+        &mut self,
+        n: u32
+    ) -> PyResult<()> {
+        if n == 0 {
+            return Ok(());
+        }
+
+        if (n as usize) > self.prior_states.len() {
+            return Err(
+                PyErr::new::<PyValueError, _>(format!(
+                    "Cannot rollback {} states. Make sure you have advanced through at least \
+                    {} states since the last rollback or object initialization",
+                    n,
+                    n
+                ))
+            );
+        }
+
+        let mut new_state: u32 = self.state;
+        for _ in 0..n {
+            new_state = self.prior_states
+                .pop_back()
+                .expect("length checked above");
```

Comment:
We need a better error message "Attempted to rollback X times but the maximum etc", basically the message above.